src/point.js

```diff
@@ -410,29 +411,29 @@ function joinLeft(joinPoint) {
   assert(joinPoint.type === BEFORE || joinPoint.type === AFTER);
   joinPoint = joinPoint.leftNormalized();
 
-  source = joinPoint.node.nextSibling;
+  source = nextSibling(joinPoint.node);
   dest = joinPoint.node;
-  parentNode = dest.parentNode;
+  prntNode = parentNode(dest);
 
 
   if (!source) {
     return Point.after(joinPoint.node);
   }
 
-  if (!dest.hasChildNodes()) {
+  if (!hasChildNodes(dest)) {
     resultPoint = Point.start(dest);
-  } else if (dest.lastChild.nodeType === 3) {
-    resultPoint = Point.text(dest.lastChild, dest.lastChild.textContent.length);
+  } else if (lastChild(dest).nodeType === 3) {
+    resultPoint = Point.text(lastChild(dest), lastChild(dest).textContent.length);
   } else {
-    resultPoint = Point.after(dest.lastChild);
+    resultPoint = Point.after(lastChild(dest));
   }
 
   if (joinPoint.type === AFTER) {
-    while(source.hasChildNodes()) {
-      var node = source.firstChild;
+    while(hasChildNodes(source)) {
+      var node = firstChild(source);
       dest.appendChild(node);
     }
-    parentNode.removeChild(source);
+    prntNode.removeChild(source);
```

Comment:
Seeing as we're using this filtered view of the dom. It seems that it would be possible for `source` to not be a child or `prntNode` which would result in an error here.